Code/Framework/AzToolsFramework/Tests/Prefab/PrefabUndoComponentPropertyOverrideTests.cpp

```diff
@@ -0,0 +1,287 @@
+
+        // Modify the IsStatic property in transform component as override
+        // Note: Entity needs to be modified manually since we would update cached instance DOM in undo node to avoid reloading
+        AZ::TransformBus::Event(wheelEntityInfo.m_entityId, &AZ::TransformInterface::SetIsStaticTransform, overriddenStaticValue);
+
+        PropertyChangePatch changePatch = MakeTransformStaticPropertyChangePatch(wheelEntityInfo.m_entityId, overriddenStaticValue);
+
+        PrefabDom propertyDomValue;
+        ConvertToPrefabDomValue(propertyDomValue, changePatch.m_propertyValue);
+
+        AZStd::string wheelEntityAliasPath = m_instanceToTemplateInterface->GenerateEntityAliasPath(wheelEntityInfo.m_entityId);
+        AZ::Dom::Path pathToPropertyFromOwningPrefab = AZ::Dom::Path(wheelEntityAliasPath) / changePatch.m_pathToProperty;
+
+        // Create an undo node
+        PrefabUndoComponentPropertyOverride undoNode("Modify transform static");
+
+        InstanceOptionalReference carInstance = m_instanceEntityMapperInterface->FindOwningInstance(carInstanceInfo.m_containerEntityId);
+        EXPECT_TRUE(carInstance.has_value());
+
+        // Redo
+        undoNode.CaptureAndRedo(carInstance->get(), pathToPropertyFromOwningPrefab, propertyDomValue);
+        EXPECT_TRUE(undoNode.Changed());
+        PropagateAllTemplateChanges();
+
+        bool currentStaticValue = defaultStaticValue; // default to opposite value
+        AZ::TransformBus::EventResult(currentStaticValue, wheelEntityInfo.m_entityId, &AZ::TransformBus::Events::IsStaticTransform);
+        EXPECT_EQ(currentStaticValue, overriddenStaticValue);
+        EXPECT_TRUE(
+            m_prefabOverridePublicInterface->AreOverridesPresent(wheelEntityInfo.m_entityId, changePatch.m_pathToProperty.ToString()));
+
+        // Undo
+        undoNode.Undo();
+        PropagateAllTemplateChanges();
+
+        currentStaticValue = overriddenStaticValue; // reset to opposite value
+        AZ::TransformBus::EventResult(currentStaticValue, wheelEntityInfo.m_entityId, &AZ::TransformBus::Events::IsStaticTransform);
+        EXPECT_EQ(currentStaticValue, defaultStaticValue);
+        EXPECT_FALSE(
+            m_prefabOverridePublicInterface->AreOverridesPresent(wheelEntityInfo.m_entityId, changePatch.m_pathToProperty.ToString()));
+
+        // Redo
+        undoNode.Redo();
+        PropagateAllTemplateChanges();
+
+        currentStaticValue = defaultStaticValue; // reset to opposite value
+        AZ::TransformBus::EventResult(currentStaticValue, wheelEntityInfo.m_entityId, &AZ::TransformBus::Events::IsStaticTransform);
+        EXPECT_EQ(currentStaticValue, overriddenStaticValue);
+        EXPECT_TRUE(
+            m_prefabOverridePublicInterface->AreOverridesPresent(wheelEntityInfo.m_entityId, changePatch.m_pathToProperty.ToString()));
+    }
+
+    TEST_F(PrefabUndoComponentPropertyOverrideTests, EditTransformStaticNoOverrideForUnchangedDefaultValue)
```

Comment:
I recommend adding a comment here to make it clearer. Something like changing 'false (default)' to 'false'